src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,197 @@
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO car (model, manufacturer_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2,car.getManufacturer().getId());
+            preparedStatement.execute();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            insertDrivers(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create field in car table with car: "
+                    + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id AS car_id, c.model AS car_model, "
+                + "m.id AS manufacturer_id, m.name AS manufacturer_name, "
+                + "m.country AS manufacturer_country "
+                + "FROM car c JOIN manufacturers m on m.id = c.manufacturer_id "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return Optional
+                    .of(getCarWithManufacturerFromResultSet(preparedStatement.executeQuery()));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get field from car table with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, c.model AS car_model, "
+                + "m.id AS manufacturer_id, m.name AS manufacturer_name, "
+                + "m.country AS manufacturer_country "
+                + "FROM car c "
+                + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarWithManufacturerFromResultSet(resultSet));
+            }
+            for (Car car : cars) {
+                car.setDrivers(getDriversForCar(car));
+            }
```

Comment:
move this logic outside the try-catch block, it is not very good to create nested connections.
or, reuse connection